### DISCUSSION TITLE
Why is Web3Modal type not exported?

### DISCUSSION THREAD

**Discussion by zaibon** [WalletConnect/walletconnect-monorepo#4333] [Developer Support] [answered]
I'm integrating web3modal into my project. I'm using typescript so I'm trying give a type to all my variable.
But it seems the type return by the `createWeb3Modal` function is not exported. Is there any specific reason for it?

**Comment by glitch-txs** [comment, marked as answer]
Feel free to open a PR for this, I thought we do. In the meantime you can do 

```ts
type Web3Modal = ReturnType<typeof createWeb3Modal>
```

**Reply by zaibon** [reply]
Great thanks, I'll provide the PR.

**Reply by zaibon** [reply]
PR is there: https://github.com/WalletConnect/web3modal/pull/2051
